Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLConvertToQ.py

```diff
@@ -0,0 +1,229 @@
+                                                     direction=Direction.Output),
+                             doc='The input workspace in momentum transfer.')
+        self.declareProperty(Prop.SUBALG_LOGGING,
+                             defaultValue=SubalgLogging.OFF,
+                             validator=StringListValidator([SubalgLogging.OFF, SubalgLogging.ON]),
+                             doc='Enable or disable child algorithm logging.')
+        self.declareProperty(Prop.CLEANUP,
+                             defaultValue=common.WSCleanup.ON,
+                             validator=StringListValidator([common.WSCleanup.ON, common.WSCleanup.OFF]),
+                             doc='Enable or disable intermediate workspace cleanup.')
+        self.declareProperty(MatrixWorkspaceProperty(Prop.REFLECTED_WS,
+                                                     defaultValue='',
+                                                     direction=Direction.Input,
+                                                     validator=WorkspaceUnitValidator('Wavelength')),
+                             doc='A non-summed reflected beam workspace, needed for Q resolution calculation.')
+        self.declareProperty(MatrixWorkspaceProperty(Prop.DIRECT_WS,
+                                                     defaultValue='',
+                                                     direction=Direction.Input,
+                                                     validator=WorkspaceUnitValidator('Wavelength')),
+                             doc='A non-summed direct beam workspace, needed for Q resolution calculation.')
+        self.declareProperty(Prop.POLARIZED,
+                             defaultValue=False,
+                             doc='True if input workspace has been corrected for polarization efficiencies.')
+        self.declareProperty(Prop.GROUPING_FRACTION,
+                             defaultValue=Property.EMPTY_DBL,
+                             validator=positiveFloat,
+                             doc='If set, group the output by steps of this fraction multiplied by Q resolution')
+
+    def _convertToMomentumTransfer(self, ws):
+        """Convert the X units of ws to momentum transfer."""
+        reflectedWS = self.getProperty(Prop.REFLECTED_WS).value
+        reflectedForeground = self._foreground(reflectedWS.run())
+        directWS = self.getProperty(Prop.DIRECT_WS).value
+        directForeground = self._foreground(directWS.run())
+        logs = ws.run()
+        instrumentName = ws.getInstrument().getName()
+        if instrumentName != 'D17' and instrumentName != 'Figaro':
+            raise RuntimeError('Unrecognized instrument {}. Only D17 and Figaro are supported.'.format(instrumentName))
+        sumType = logs.getProperty(common.SampleLogs.SUM_TYPE).value
+        polarized = self.getProperty(Prop.POLARIZED).value
+        pixelSize = 0.001195 if instrumentName == 'D17' else 0.0012
```

Comment:
They are not likely to change, expect pixel size and resolution get smaller when they get more accurate estimates. Anyway, some of these values are actually needed also elsewhere, so IPF is a good option. I've opened issue #22470 to track the progress since this change is not urgent.